addons/mail/models/mail_partner.py

```diff
@@ -0,0 +1,132 @@
+        add partner registration and auto subscribe of partner on reply.
+        -Register mail informations on "new_message" (email_from,email_cc,partner_id,name)
+        -Create partner on update_message or new_message with respect to the "message_subscription_strategy".
+        -The list of partner to subcribe and/or register are defined by "emails_to_subscribe" 
+        and "partner_temails_to_subscribe". By default, only the "email_from" used for the creation of the record 
+        will be subscribed. 
+        -Once a partner is created, (using odoo interface when posting message or when repplying to a mail), 
+        the field partner_id (can be changed with "message_partner_id_field") is updated with the new partner 
+        corresponding to email_from for every record having the same value for "email_from" and partner field not set 
+        (partner_id or message_partner_id_field if overwritten). 
+    """
+    ON_REPLY_STRATEGY,ON_CREATE_STRATEGY,NEVER_STRATEGY = range(3)
+    _name = 'mail.partner.mixin'
+    
+    email_from = fields.Char('Email', help="Email address of the contact", index=True)
+    email_cc = fields.Text('Global CC', help="These email addresses will be added to the CC field of all inbound and outbound emails for this record before being sent. Separate multiple email addresses with a comma")
+    partner_id = fields.Many2one('res.partner', string='Customer', track_visibility='onchange', index=True, help="Linked partner (optional). Usually created when converting the lead.")
+    name = fields.Char("Subject", translate=True)
+
+    @api.model
+    def message_new(self, msg, custom_values=None):
+        """ Overrides mail_thread new that is called by the mailgateway
+            through process.
+            This override adds the name, email_from, email_cc and partner_id to the email.
+        """
+        # remove default author when going through the mail gateway. Indeed we
+        # do not want to explicitly set user_id to False; however we do not
+        # want the gateway user to be responsible if no other responsible is
+        # found.
+        create_context = dict(self.env.context or {})
+        create_context['default_user_id'] = False
+        if custom_values is None:
+            custom_values = {}
+        defaults = {
+            'name': msg.get('subject') or _("No Subject"),
+            'email_from': msg.get('from'),
+            'email_cc': msg.get('cc'),
+            'partner_id': msg.get('author_id', False),
+        }
+        defaults.update(custom_values)
+        thread = super(MailPartnerMixin, self.with_context(create_context)).message_new(msg, custom_values=defaults)
+        if(thread.message_subscription_strategy(msg) == self.ON_CREATE_STRATEGY): #if we want to subscribe emails on create
+            thread.partner_auto_subscribe(msg)
+        return thread
+
+    def message_update(self, msg, update_vals=None):
```

Comment:
Side note: we still have some message update override to support commands in incoming emails. Couldn't we get rid of them ?

Reply:
I think only projects and crm lead are concerned. The main operations in common are to read the body and search for commands to update fields. We could think about adding this logic to the mixin (a little out of topeic when talking about the MailPartnerMixin, would need to rename that part) or to create another Mixin to manage this logic.